contracts/MarketOracle.sol

```diff
@@ -72,13 +105,34 @@ contract MedianOracle is Ownable, IOracle {
         uint256[] memory validReports = new uint256[](reportsCount);
         uint256 size = 0;
         uint256 minValidTimestamp =  now.sub(reportExpirationTimeSec);
+        uint256 maxValidTimestamp =  now.sub(reportSecurityDelaySec);
 
         for (uint256 i = 0; i < reportsCount; i++) {
             address providerAddress = providers[i];
-            if (minValidTimestamp <= providerReports[providerAddress].timestamp) {
-                validReports[size++] = providerReports[providerAddress].payload;
+            Report[2] memory reports = providerReports[providerAddress];
+            uint8 index_recent;
+            if (reports[0].timestamp >= reports[1].timestamp) {
+                index_recent = 0;
             } else {
-                emit ReportExpired(providerAddress);
+                index_recent = 1;
+            }
+            uint8 index_older = 1 - index_recent;
+
+            if (reports[index_recent].timestamp <= maxValidTimestamp) { // Not too recent
```

Comment:
Sometimes we use strict comparators and sometimes `>=` style comparators. Is there a reason for the inconsistency?

Reply:
Yes, the valid range is [minValidTimestamp, maxValidTimestamp] inclusive. When checking if the a timestamp is within valid range >= <= are used if checking if it is outside > < are used. Will see if I can restructure it in a way where they are all the same.

Reply:
Got rid of the inconsistency and cleaned the structure.

Reply:
Like it!